cranelift/codegen/src/opts/cprop.isle

```diff
@@ -264,3 +264,18 @@
 (extern constructor f32_from_sint f32_from_sint)
 (decl f64_from_sint (i64) Ieee64)
 (extern constructor f64_from_sint f64_from_sint)
+
+;; Constant fold bswap of a constant.
+(rule (simplify (bswap (ty_16 ty) (iconst ty (u64_from_imm64 n))))
+      (subsume (iconst ty (imm64 (u64_bswap16 n)))))
+(rule (simplify (bswap (ty_32 ty) (iconst ty (u64_from_imm64 n))))
+      (subsume (iconst ty (imm64 (u64_bswap32 n)))))
+(rule (simplify (bswap (ty_64 ty) (iconst ty (u64_from_imm64 n))))
+      (subsume (iconst ty (imm64 (u64_bswap64 n)))))
```

Comment:
Instead of the `ty_NN` matchers I think you can match `$I16`, `$I32`, and `$I64` here directly which would also avoid the need for a new `ty_16` (not that it's wrong, just perhaps a nice small cleanup)

Reply:
Aha, thanks, I knew I was missing something!